internal/kldrest/receiptstore_test.go

```diff
@@ -187,6 +199,29 @@ func TestReplyProcessorWithPeristenceErrorSwallows(t *testing.T) {
 	})
 }
 
+func TestReplyProcessorWithPeristenceErrorDuplicateSwallows(t *testing.T) {
+	existing := map[string]interface{}{"some": "existing"}
+	r := newReceiptStore(&ReceiptStoreConf{
+		RetryTimeoutMS:      1,
+		RetryInitialDelayMS: 1,
+	}, &mockReceiptErrs{
+		addReceiptErr: fmt.Errorf("pop"),
+		getReceiptErr: nil,
+		getReceiptVal: &existing,
+	}, nil)
+
+	replyMsg := &kldmessages.TransactionReceipt{}
+	replyMsg.Headers.MsgType = kldmessages.MsgTypeTransactionSuccess
+	replyMsg.Headers.ID = kldutils.UUIDv4()
+	replyMsg.Headers.ReqID = kldutils.UUIDv4()
+	replyMsg.Headers.ReqOffset = "topic:1:2"
+	txHash := common.HexToHash("0x02587104e9879911bea3d5bf6ccd7e1a6cb9a03145b8a1141804cebd6aa67c5c")
+	replyMsg.TransactionHash = &txHash
+	replyMsgBytes, _ := json.Marshal(&replyMsg)
+
+	r.processReply(replyMsgBytes)
```

Comment:
It would be good to validate the correct behavior (the mock?) was called in this particular case. It would give me more confidence that the new behavior flows as expected.

Reply:
👍  - added a verification that the error had been thrown during the test.